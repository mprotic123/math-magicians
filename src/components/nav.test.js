import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import Nav from './nav.js';

it('renders nav', () => {
  const nav = renderer.create(
    <Router>
    <Nav />
    </Router>,
  ).toJSON();
  expect(nav).toMatchSnapshot();
});

test('Navbar calculator button function', () => {
  render(<Router>
      <Nav />
      </Router>);
  const calcLink = screen.getByText('Calculator');
  const history = createMemoryHistory();

  userEvent.click(calcLink);

  expect(history.location.pathname).toBe('/');
});