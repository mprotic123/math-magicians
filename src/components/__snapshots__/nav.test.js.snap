// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders nav 1`] = `
<nav>
  <h1>
    Math Magicians
  </h1>
  <ul
    className="navContain"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <li
        className="nav-item"
      >
        Home
      </li>
    </a>
    <a
      href="/calc"
      onClick={[Function]}
    >
      <li
        className="nav-item"
        data-testid="calculator"
      >
        Calculator
      </li>
    </a>
    <a
      href="/quote"
      onClick={[Function]}
    >
      <li
        className="nav-item"
      >
        Quote
      </li>
    </a>
  </ul>
</nav>
`;
